src/Mgate/SuiviBundle/Controller/EtudeController.php

```diff
@@ -232,12 +229,22 @@ public function modifierAction(Request $request, Etude $etude)
                 $em->persist($etude);
                 $em->flush();
 
-                return $this->redirect($this->generateUrl('MgateSuivi_etude_voir', array('nom' => $etude->getNom())));
+                if ($request->get('ap')) {
+                    return $this->redirect($this->generateUrl('MgateSuivi_ap_rediger', array('id' => $etude->getId())));
+                } else {
+                    return $this->redirect($this->generateUrl('MgateSuivi_etude_voir', array('nom' => $etude->getNom())));
+                }
+            } else {
+                $errors = $this->get('validator')->validate($etude);
+                foreach ($errors as $error) {
+                    array_push($error_messages, $error->getPropertyPath() . ' : ' . $error->getMessage());
```

Comment:
I would rather use flashbags `$this->addFlash()` than recreating the same mecganism with a custom array